extensions/devservices/db2/src/main/java/io/quarkus/devservices/db2/deployment/DB2DevServicesProcessor.java

```diff
@@ -57,4 +53,53 @@ public void close() throws IOException {
         });
     }
 
+    private static class QuarkusDb2Container extends Db2Container {
+        private final OptionalInt fixedExposedPort;
+        private final boolean useSharedNetwork;
+
+        private String hostName = null;
+
+        public QuarkusDb2Container(Optional<String> imageName, OptionalInt fixedExposedPort, boolean useSharedNetwork) {
+            super(DockerImageName.parse(imageName.orElse("ibmcom/db2:" + DB2DevServicesProcessor.TAG))
+                    .asCompatibleSubstituteFor(DockerImageName.parse("ibmcom/db2")));
+            this.fixedExposedPort = fixedExposedPort;
+            this.useSharedNetwork = useSharedNetwork;
+        }
+
+        @Override
+        protected void configure() {
+            super.configure();
+
+            if (useSharedNetwork) {
```

Comment:
Is there any chance this code can be moved to a shared utility method so it is not copied everywhere?

Reply:
I'll look into it

Reply:
I moved the shared network stuff into a utility method. The rest isn't that straighforward as some of the methods are protected